### ISSUE TITLE
late joining permanent peers cannot join a bastion ring

### ISSUE THREAD

**Issue by mwrock** [habitat-sh/habitat#7510]
When setting up a bastion ring, if all permanent peers come up within about a minute, that all seem to join and all is well. However if a peer does not join before the other peers get to the p;oint where their debug output reports that they are waiting `until the next protocol period` then it will never join that ring. Restarting one or all peers has no effect. The only remedy seems to be to delete `/hab/sup/default` on all peers and restart.

I have not done much diving in the code yet but that is my next stage of investigation.

**Comment by mwrock** [comment]
looking through the butterfly code, we start the server and check to see if its count of members is at least half+1 of all gossip peer args(initial members). That is 2 for a 3 node bastion ring. if not, we send ping requests to these initial members and wait. assuming one peer has not come on line yet, it will only get an ack from that one, add it to the member list and now it has 2 members because we can assume itself is running also as the 2nd member. Now it has 2 members and it will never ever check the initial_member list again. Now that these 2 members have been persisted to the .dat file they will be read in on subsequent restarts satisfying the minimum number of 2 members needed to proceed.

The key code I am looking at is https://github.com/habitat-sh/habitat/blob/4f0666141a16e42272324dd5e74a653146cbb526/components/butterfly/src/server/outbound.rs#L83

**Comment by mwrock** [comment]
Another workaround which is totally awful is juat add 2 bogus IPs to one of the successfully joined servers to bump the minimum threshold and that missing peer is accepted into the fold

**Comment by mwrock** [comment]
Its actually looking like this is not a bug but an environmental issue. I set up a 3 container permanent peer network in an attempt to repro this issue quickly. In this docker based environment I was not able to reproduce the problem at all. I could only reproduce it in my 3 EC2 VM environment.

Looking closer at the code my assumptions about the nature of this bug seem to be false. Even after a node identifies half of its initial peers and starts probing those peers that have joined. It should still acknowledge a ping from a late joining peer and add it to the member list.

After adding some debug output, it looks like the established members just were not receiving the UDP ping of the late peer trying to join.. Looking at a packet trace, I could see the established 2 members exchanging their UDP packets and I could see the late peer sending UDP pings but the trace on the established peers were not receiving the packets in their trace. That seemed really weird.

I then looked at the EC2 security group and noted no rule allowing UDP traffic on port 9638 was added. I added it and then the late peer was immediately joined to the supervisor network. What baffles me here is how the UDP packets were being exchanged among the peers that joined at the same time when this rule was not added to the security group?! I'd expect them to be blocked.
